<a href="https://colab.research.google.com/github/SURESHASWATHI/assignment/blob/main/HYPOTHESIS_TESTING.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Task
Perform a hypothesis test to investigate if the weekly operating costs for Bombay Hospitality Ltd. franchises are higher than the theoretical model W = $1,000 + $5X suggests, given a sample of 25 restaurants with a mean weekly cost of Rs. 3,050, where X follows a normal distribution with a mean of 600 units and a standard deviation of 25 units, using an alpha level of 5%. State the hypotheses, calculate the test statistic, determine the critical value, make a decision, and provide a conclusion.

## State the hypotheses

### Subtask:
Define the null and alternative hypotheses based on the problem description.


**Reasoning**:
Define and state the null and alternative hypotheses based on the problem description.



In [1]:
# Null Hypothesis (H0): The mean weekly operating cost is equal to or less than the cost predicted by the theoretical model.
# H0: mu <= 1000 + 5 * 600  => mu <= 4000

# Alternative Hypothesis (Ha): The mean weekly operating cost is higher than the cost predicted by the theoretical model.
# Ha: mu > 1000 + 5 * 600 => mu > 4000

print("Null Hypothesis (H0): The mean weekly operating cost is less than or equal to 4000.")
print("Alternative Hypothesis (Ha): The mean weekly operating cost is greater than 4000.")

Null Hypothesis (H0): The mean weekly operating cost is less than or equal to 4000.
Alternative Hypothesis (Ha): The mean weekly operating cost is greater than 4000.


## Calculate the theoretical mean weekly cost

### Subtask:
Use the given cost model and the mean number of units produced (X) to calculate the theoretical mean weekly cost (μ).


**Reasoning**:
Assign the given values and calculate the theoretical mean weekly cost.



In [2]:
fixed_cost = 1000
variable_cost_per_unit = 5
mean_x = 600

theoretical_mean_cost = fixed_cost + variable_cost_per_unit * mean_x

print(f"The theoretical mean weekly cost (μ) is: {theoretical_mean_cost}")

The theoretical mean weekly cost (μ) is: 4000


## Calculate the standard deviation

### Subtask:
Calculate the standard deviation of the weekly cost (σ) based on the given information.


**Reasoning**:
Calculate the standard deviation of the weekly cost by multiplying the standard deviation of X by the variable cost per unit.



In [3]:
std_dev_x = 25
std_dev_w = variable_cost_per_unit * std_dev_x
print(f"The standard deviation of the weekly cost (σ) is: {std_dev_w}")

The standard deviation of the weekly cost (σ) is: 125


## Calculate the test statistic

### Subtask:
Use the provided formula, the sample mean, the theoretical mean, the standard deviation, and the sample size to calculate the test statistic (t).


**Reasoning**:
Calculate the standard error and the test statistic using the provided values and formulas.



In [4]:
import numpy as np

sample_mean_cost = 3050
sample_size = 25

# standard_dev_w is already calculated in a previous step

standard_error = std_dev_w / np.sqrt(sample_size)
test_statistic = (sample_mean_cost - theoretical_mean_cost) / standard_error

print(f"The calculated test statistic (t) is: {test_statistic}")

The calculated test statistic (t) is: -38.0


## Determine the critical value

### Subtask:
Find the critical value from the standard normal (Z) distribution table for a given alpha level (α = 0.05).


**Reasoning**:
Determine the critical Z-value for a one-tailed test with alpha = 0.05.



In [5]:
from scipy.stats import norm

# The alternative hypothesis (Ha: mu > 4000) indicates a one-tailed test (right-tailed).
alpha = 0.05

# For a right-tailed test with alpha = 0.05, we need the Z-value that leaves 0.05 in the upper tail.
# This is equivalent to finding the Z-value that leaves 1 - alpha = 0.95 in the lower tail.
critical_z_value = norm.ppf(1 - alpha)

print(f"The critical Z-value for a one-tailed test with alpha = {alpha} is: {critical_z_value}")

The critical Z-value for a one-tailed test with alpha = 0.05 is: 1.6448536269514722


## Make a decision

### Subtask:
Compare the calculated test statistic with the critical value to decide whether to reject the null hypothesis.


**Reasoning**:
Compare the calculated test statistic with the critical Z-value and print the decision regarding the null hypothesis.



In [6]:
print(f"Calculated Test Statistic: {test_statistic}")
print(f"Critical Z-value: {critical_z_value}")

if test_statistic > critical_z_value:
  print("Decision: Reject the null hypothesis.")
  print("Reason: The test statistic is greater than the critical Z-value.")
else:
  print("Decision: Do not reject the null hypothesis.")
  print("Reason: The test statistic is less than or equal to the critical Z-value.")

Calculated Test Statistic: -38.0
Critical Z-value: 1.6448536269514722
Decision: Do not reject the null hypothesis.
Reason: The test statistic is less than or equal to the critical Z-value.


## Conclusion

### Subtask:
Based on the decision, write a conclusion about whether there is evidence to support the restaurant owners' claim.


**Reasoning**:
Based on the decision from the previous step (not rejecting the null hypothesis), write a conclusion that addresses the restaurant owners' claim.



In [7]:
# Recall the hypotheses:
# H0: mu <= 4000 (Mean weekly operating cost is less than or equal to the theoretical model suggests)
# Ha: mu > 4000 (Mean weekly operating cost is higher than the theoretical model suggests)

# Recall the decision: Do not reject the null hypothesis.

print("Conclusion:")
print("Based on the hypothesis test, we did not reject the null hypothesis (H0).")
print("This means there is not enough statistical evidence at the 5% significance level")
print("to support the restaurant owners' claim that the weekly operating costs are higher")
print("than the theoretical model ($1,000 + $5X) suggests.")

Conclusion:
Based on the hypothesis test, we did not reject the null hypothesis (H0).
This means there is not enough statistical evidence at the 5% significance level
to support the restaurant owners' claim that the weekly operating costs are higher
than the theoretical model ($1,000 + $5X) suggests.


## Summary:

### Data Analysis Key Findings

*   The theoretical mean weekly operating cost based on the model W = \$1,000 + \$5X and a mean X of 600 units is \$4,000.
*   The standard deviation of the weekly operating cost (W) is calculated as 5 times the standard deviation of X (25), resulting in \$125.
*   The calculated test statistic for the sample of 25 restaurants with a mean weekly cost of \$3,050 is -38.0.
*   For a one-tailed hypothesis test with an alpha level of 0.05, the critical Z-value is approximately 1.645.
*   The test statistic (-38.0) is less than the critical Z-value (1.645), leading to the decision to not reject the null hypothesis.

### Insights or Next Steps

*   The analysis indicates that the sample data does not provide sufficient evidence at the 5% significance level to conclude that the weekly operating costs are higher than the theoretical model suggests.
*   Given the significantly lower sample mean compared to the theoretical mean, further investigation into potential factors causing this difference (e.g., efficiencies in operations, lower actual unit production than the mean X, or inaccuracies in the theoretical model itself) could be beneficial.
